spec/javascripts/angular/controllers/dashboard_ctrl_spec.js

```diff
@@ -0,0 +1,55 @@
+describe('DashboardCtrl', function() {
+  var $controller;
+  var $scope;
+  beforeEach(inject(function(_$controller_, $rootScope){
+    $controller = _$controller_;
+    $scope = $rootScope.$new(true);
+  }));
+
+  describe('visual toggles', function() {
+    describe('clone modal', function() {
+      beforeEach(function() {
+        $controller('DashboardCtrl', { $scope: $scope });
+      });
+
+      it('starts closed', function() {
+        expect($scope.showCloneControls).toEqual(false);
+      });
+
+      it('opens the controls', function() {
+        $scope.showCloneMenu();
+        expect($scope.showCloneControls).toEqual(true);
+      });
+
+      it('can close the modal', function() {
+        expect($scope.showCloneControls).toEqual(false);
+        $scope.showCloneMenu();
+        expect($scope.showCloneControls).toEqual(true);
+        $scope.closeCloneControls();
+        expect($scope.showCloneControls).toEqual(false);
+      });
+    });
+
+    describe('fullscreen', function() {
+      beforeEach(function() {
+        $controller('DashboardCtrl', { $scope: $scope });
+      });
+
+      it('starts not in fullscreen', function() {
+        expect($scope.fullscreen).toEqual(false);
+      });
+
+      it('toggles to fullscreen', function() {
+        $scope.enableFullscreen();
+        expect($scope.fullscreen).toEqual(true);
```

Comment:
Do you think it would make sense to add another validation here to verify that fulltext also changes the UI? At the moment the text basically covers that `enableFullscreen` and `exitFullscreen` change the `fullscreen` variable between true and false. It might be good to have a test somewhere to ensure that it removes the navigation bar, form elements and makes the graphs bigger or so.